Editor/src/Editor.bf

```diff
@@ -0,0 +1,615 @@
+			String fileName = scope .(themeName)..Append(".txt");
+			String _themePath = scope .();
+			SteelPath.GetEditorResourcePath(_themePath, "Themes", fileName);
+
+			String theme = new .();
+			defer delete theme;
+
+			File.ReadAllText(_themePath, theme);
+			LoadStyle(theme);
+		}
+
+		public static void GetEntityName(EntityId id, String buffer)
+		{
+			var editor = GetInstance<Editor>();
+			if (!editor._entityNames.ContainsKey(id))
+				SetEntityName(id, "Entity");
+
+			buffer.Append(editor._entityNames[id]);
+		}
+
+		public static void SetEntityName(EntityId id, StringView name)
+		{
+			var editor = GetInstance<Editor>();
+			if (!editor._entityNames.ContainsKey(id))
+				editor._entityNames[id] = new .(name);
+			else
+				editor._entityNames[id].Set(name);
+		}
+
+		public static void Refresh()
+		{
+			for (var window in GetInstance<Editor>()._editorLayer.[Friend]_editorWindows)
+				window.OnShow();
+		}
+
+		public static void OpenProject(StringView path)
+		{
+			GameConsole.Instance.Clear();
+
+			var filePath = scope String();
+			Path.InternalCombine(filePath, scope String(path), "SteelProj.json");
+
+			if (!File.Exists(filePath))
+			{
+				Log.Error("Could not open project ({}): Not a Steel project", path);
+				return;
+			}
+
+			var json = new String();
+			defer delete json;
+
+			if (File.ReadAllText(filePath, json) case .Err(let err))
+			{
+				Log.Error("Could not open project ({}): {}", path, err);
+				return;
+			}
+
+			var editor = GetInstance<Editor>();
+			delete editor.CurrentProject;
+			editor.CurrentProject = new .();
+
+			for (var entity in Entity.EntityStore.Values)
+				delete entity;
+			Entity.EntityStore.Clear();
+
+			if (JSONDeserializer.Deserialize<EditorProject>(json, editor.CurrentProject) case .Err(let err))
+			{
+				Log.Error("Could not open project ({}): {}", path, err);
+				return;
+			}
+
+			for (var serializableEntity in editor.CurrentProject.Entities)
+				serializableEntity.MakeEntity();
+
+			editor.CurrentProject.Path = new .(path);
+			UpdateTitle();
+
+			editor._cache.AddRecentProject(path);
+
+			InspectorWindow.SetCurrentEntity(null);
+			SteelPath.SetContentDirectory();
+			Refresh();
+		}
+
+		public static void CloseProject()
+		{
+			Log.Info("Closing project");
+
+			var editor = GetInstance<Editor>();
+			delete editor.CurrentProject;
+			editor.CurrentProject = EditorProject.UntitledProject();
+			UpdateTitle();
+		}
+
+		public static void UpdateTitle()
+		{
+			var editor = GetInstance<Editor>();
+			var title = scope String();
+
+			if (editor.CurrentProject.Path.IsEmpty)
+				title.AppendF("Steel Editor - {}", editor.CurrentProject.Name);
+			else
+				title.AppendF("Steel Editor - {}", editor.CurrentProject.Path);
+
+			if (editor._wantsSave)
+				title.Append('*');
+
+			editor.Window.SetTitle(title);
+		}
+
+		public static void Save()
+		{
+			Log.Trace("Saving project");
+
+			var editor = GetInstance<Editor>();
+
+			editor.CurrentProject.Entities.Clear();
```

Comment:
This will certainly cause memory leak as the stored serializable entities are never freed.